### ISSUE TITLE
[java] CloseResource: support Lombok's @Cleanup annotation

### ISSUE THREAD

**Issue by TomerFi** [pmd/pmd#2757]
**Affects PMD Version:** 

[6.27.0](https://github.com/pmd/pmd/releases/tag/pmd_releases%2F6.27.0)

**Description:**

Version [6.27.0](https://github.com/pmd/pmd/releases/tag/pmd_releases%2F6.27.0) breaks support for [Lombok's Cleanup Annotation](https://projectlombok.org/features/Cleanup) which generates invocation of the `close()` method for closable objects.

Falling back to [6.26.0](https://github.com/pmd/pmd/releases/tag/pmd_releases%2F6.26.0) works as expected by not producing the same violation.

**Exception Stacktrace:**

```xml
<violation ... rule="CloseResource" ruleset="Error Prone" ... class="ShabbatTimesStreamHandler" method="getCreator" variable="context" externalInfoUrl="https://pmd.github.io/pmd-6.27.0/pmd_rules_java_errorprone.html#closeresource" priority="3">
Ensure that resources like this AnnotationConfigApplicationContext object are closed after use
</violation>
```

**Code Sample demonstrating the issue:**

Copied from a public repository: [TomerFi/alexa-skill-shabbat-times](https://github.com/TomerFi/alexa-skill-shabbat-times/blob/master/src/main/java/info/tomfi/alexa/skills/shabbattimes/ShabbatTimesStreamHandler.java)

```java
public class ShabbatTimesStreamHandler extends SkillStreamHandler {
  private static ShabbatTimesSkillCreator getCreator() {
    @Cleanup val context = new AnnotationConfigApplicationContext(DiProdConfiguration.class);
    return context.getBean(ShabbatTimesSkillCreator.class);
  }

  public ShabbatTimesStreamHandler() {
    super(getCreator().getSkill());
  }
}
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  clone the repository:
```shell
git clone https://github.com/TomerFi/alexa-skill-shabbat-times.git
```

2. step into the cloned folder:
```shell
cd alexa-skill-shabbat-times
```

3. invoke maven goal (no violation is detected  with version 6.26.0):
```shell
mvn pmd:check
```

4. update the `pmd.version` property in `pom.xml` to 6.27.0 to catch the violation:
```xml
  <properties>
   ...
    <pmd.version>6.27.0</pmd.version>
   ...
  </properties>
```

5. invoke maven goal
```shell
mvn pmd:check
```
will now produce:
```text
You have 1 PMD violation.
```

**Running PMD through:** *Maven*

**Comment by acranmer** [comment]
I noticed it broke for me with 6.16.0

**Comment by oowekyala** [comment]
I wonder if this was fixed by #3157

**Comment by oowekyala** [comment]
After investigating, this problem is still there on master. But that's because the rule never had support for lombok's `@Cleanup` in the first place. So, this needs to be added to the rule.
